src/openswathalgo/include/OpenMS/OPENSWATHALGO/ALGO/MRMScoring.h

```diff
@@ -95,44 +95,67 @@ namespace OpenSwath
     void initializeXCorrMatrix(OpenSwath::IMRMFeature* mrmfeature, const std::vector<String>& native_ids);
 
-    /// Initialize the cross-correlation vector with the MS1 trace
-    void initializeMS1XCorr(OpenSwath::IMRMFeature* mrmfeature, const std::vector<String>& native_ids, const std::string& precursor_id);
+    /// Initialize the scoring object and building the cross-correlation matrix of chromatograms of set1 (e.g. identification transitions) vs set2 (e.g. detection transitions)
+    void initializeXCorrContrastMatrix(OpenSwath::IMRMFeature* mrmfeature, const std::vector<String>& native_ids_set1, const std::vector<String>& native_ids_set2);
 
-    /// Initialize the scoring object and building the cross-correlation matrix of identification vs detection chromatograms
-    void initializeXCorrIdMatrix(OpenSwath::IMRMFeature* mrmfeature, const std::vector<String>& native_ids_identification, const std::vector<String>& native_ids_detection);
+    /// Initialize the scoring object and building the cross-correlation matrix
+    void initializeXCorrPrecursorMatrix(OpenSwath::IMRMFeature* mrmfeature, const std::vector<String>& precursor_ids);
+
+    /// Initialize the scoring object and building the cross-correlation matrix of chromatograms of precursor isotopes vs transitions
+    void initializeXCorrPrecursorContrastMatrix(OpenSwath::IMRMFeature* mrmfeature, const std::vector<String>& precursor_ids, const std::vector<String>& native_ids);
 
     /// calculate the cross-correlation score
     double calcXcorrCoelutionScore();
-    std::string calcIndXcorrIdCoelutionScore();
+
+    /// calculate the weighted cross-correlation score
+    double calcXcorrCoelutionWeightedScore(const std::vector<double>& normalized_library_intensity);
+
+    /// calculate the cross-correlation contrast score
+    double calcXcorrContrastCoelutionScore();
+
+    /// calculate the separate cross-correlation contrast score
+    std::string calcSeparateXcorrContrastCoelutionScore();
+
+    /// calculate the precursor cross-correlation contrast score
+    double calcXcorrPrecursorCoelutionScore();
+
+    /// calculate the precursor cross-correlation contrast score against the transitions
+    double calcXcorrPrecursorContrastCoelutionScore();
 
     /// calculate the cross-correlation shape score
-    double calcXcorrShape_score();
-    std::string calcIndXcorrIdShape_score();
+    double calcXcorrShapeScore();
 
     /// calculate the weighted cross-correlation shape score
-    double calcXcorrShape_score_weighted(const std::vector<double>& normalized_library_intensity);
+    double calcXcorrShapeWeightedScore(const std::vector<double>& normalized_library_intensity);
 
-    /// calculate the weighted cross-correlation score
-    double calcXcorrCoelutionScore_weighted(const std::vector<double>& normalized_library_intensity);
+    /// calculate the cross-correlation contrast shape score
+    double calcXcorrContrastShapeScore();
```

Comment:
We currently use ``initializeXCorrContrastMatrix`` only for transition-level scoring in IPF. However we could now directly implement SILAC scoring in combination with the new function ``calcXcorrContrastShapeScore``.

Reply:
that would be interesting, however for SILAC scoring we would not have the second set of transitions -- but we could correlate the two MS1 traces